### ISSUE TITLE
can't   update   Or  install     ownCloud  in version  10.0.1

### ISSUE THREAD

**Issue by jonesmith** [owncloud/core#28007]
### Actual behaviour
Tell us what happens instead
```
I recently got to know owncloud, when i install   owncloud 10.0.0.  ,it is  wonderful  work.but  when  i update to owncloud 10.0.1   ,i  get  a  error   "Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE TABLE oc_account_terms (id BIGINT UNSIGNED AUTO_INCREMENT NOT NULL, account_id BIGINT UNSIGNED NOT NULL, term VARCHAR(256) NOT NULL, INDEX account_id_index (account_id), INDEX term_index (term), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes"

i know  the error  code  is   ' term varchar(256)',but  i can't  update the  owncloud  code.  so  ~ 

and  then   i   reinstall  in   owncloud 10.0.1  ,it happen  again.the  same  error.
```
### Server configuration
**Operating system**:
ubuntu 
**Web server:**
apache
**Database:**
mysql 5.6
**PHP version:**
PHP Version 5.6.30-0+deb8u1

**ownCloud version:** (see ownCloud admin page)
10.0.1
**Updated from an older ownCloud or fresh install:**
install     | same  as  update  from   10.0.0

#### Web server error 
```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE TABLE oc_account_terms (id BIGINT UNSIGNED AUTO_INCREMENT NOT NULL, account_id BIGINT UNSIGNED NOT NULL, term VARCHAR(256) NOT NULL, INDEX account_id_index (account_id), INDEX term_index (term), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes
```

#### error log
```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE TABLE oc_account_terms (id BIGINT UNSIGNED AUTO_INCREMENT NOT NULL, account_id BIGINT UNSIGNED NOT NULL, term VARCHAR(256) NOT NULL, INDEX account_id_index (account_id), INDEX term_index (term), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_bin ENGINE = InnoDB': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes
```

**Comment by PVince81** [comment]
> 1071 Specified key was too long; max key length is 767 bytes

@DeepDiver1975

**Comment by PVince81** [comment]
also cc @tomneedham @butonic as I remember a discussion about index size for this particular table

**Comment by tomneedham** [comment]
This is the column and related index https://github.com/owncloud/core/blob/f4b1dc146366595bb291411804d20000c6d7e9ca/core/Migrations/Version20170516100103.php#L51-L58

**Comment by tomneedham** [comment]
according to https://dev.mysql.com/doc/refman/5.7/en/innodb-restrictions.html we can only go up to 255 chars, and we're trying 256 😞

**Comment by jerryjharrison** [comment]
Any update on the correction to this?  I am a long time Owncloud user, but this is completely blocking the upgrade path.  I typically use pkg to update, so I unless someone has a manual correction, I suppose I have to wait for the next version release to migrate through pkg before upgrading?

**Comment by PVince81** [comment]
@tomneedham a quickfix in the form of manually executable SQL statements would be nice.

Basically you need to make the column "term" 255 characters instead of 256.

**Comment by tomneedham** [comment]
Applying this patch to the 10.0.1 codebase before upgrading will make the upgrade run smoothly:
```
--- a/core/Migrations/Version20170516100103.php
+++ b/core/Migrations/Version20170516100103.php
@@ -50,7 +50,7 @@ public function changeSchema(Schema $schema, array $options) {
 
 			$table->addColumn('term', Type::STRING, [
 				'notnull' => true,
-				'length' => 256
+				'length' => 255
 			]);
 
 			$table->setPrimaryKey(['id']);
```

**Comment by jerryjharrison** [comment]
Not familiar with this process.  Could you provide some guidance on steps required?


________________________________

**Comment by PVince81** [comment]
I'm still wondering why we didn't find this in local and automated tests. Is this a different version of Mysql / MariaDB or does it have different settings ?

**Comment by PVince81** [comment]
@jerryjharrison just edit the file core/Migrations/Version20170516100103.php and on line 52 replace the value "256" with "255".

**Comment by PVince81** [comment]
please all post your Mysql/mariadb version number and config if possible...

**Comment by jerryjharrison** [comment]
For me it is:
mysql56-server-5.6.36          Multithreaded SQL database (server)

I am happy to provide you any configuration information you want, but it was a clean install using pkg.  The only thing I did after install was to run mysql_secure_installation, and then install owncloud.  Nothing other than owncloud is installed on this computer.

**Comment by jerryjharrison** [comment]
Sorry, but I am a bit of a noob around github, and downloading the code.  I have always installed from pkg.

I realize it may be too basic for this thread, but would love to learn how to get and install from github.

**Comment by tomneedham** [comment]
> I'm still wondering why we didn't find this in local and automated tests. Is this a different version of Mysql / MariaDB or does it have different settings ?

guessing we dont have mysql strict mode enabled on our CI tests => we should turn it on

**Comment by jerryjharrison** [comment]
I just tried the upgrade to 10.0.2, and it failed due to database schema issues again…

Is anybody testing this before it is published?

**Comment by tomneedham** [comment]
@jerryjharrison We have two fixes already implemented for this issue: https://github.com/owncloud/core/pull/28019, https://github.com/owncloud/core/pull/28086

As previously mentioned, please disable strict mode if you wish to upgrade to 10.0.2, or wait for 10.0.3 with these fixes.

**Comment by tomneedham** [comment]
Will close this issue once the PR is merged in.

**Comment by tomneedham** [comment]
> Is anybody testing this before it is published?

@jerryjharrison We're an open source project, everyone is here to help, code, test. Please help out by testing the PRs mentioned above and report back with feedback - this is how we advance the project together

**Comment by jerryjharrison** [comment]
I do my best on the testing, and regularly contribute when I can.  I have reviewed all of the comments on the two open tickets, and cannot find any instructions/guidance on how to turn off strict mode on sql.  Additionally, I had requested additional help and would love to learn how to install from the tarball/github but my experience so far is limited to installing from portsnap or pkg…

**Comment by PVince81** [comment]
waiting for the last backport before closing: https://github.com/owncloud/core/pull/28576

**Comment by PVince81** [comment]
PRs were merged. Fixes will be in 10.0.3

**Comment by sleroy** [comment]
I encountered a similar issue using the Docker version of owncloud and deploying it in GCE :

The versions  I have been using :

owncloud/server : latest
dockhippie/mariadb: latest
dockhippie//redis : latest
